internal/provider/network_connection_models.go

```diff
@@ -0,0 +1,289 @@
+							}...),
+						},
+					},
+				},
+				Validators: []validator.Object{
+					objectvalidator.ExactlyOneOf(path.Expressions{
+						path.MatchRoot("aws"),
+						path.MatchRoot("google"),
+					}...),
+				},
+			},
+			"google": resourceschema.SingleNestedAttribute{
+				Optional:            true,
+				MarkdownDescription: "Google Cloud connection info",
+				Attributes: map[string]resourceschema.Attribute{
+					"name": resourceschema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "Name of this peering",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					"peer_network_url": resourceschema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "The URL of the peer network",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					"managed_network_url": resourceschema.StringAttribute{
+						Computed:            true,
+						MarkdownDescription: "The URL of the managed GCP network",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+				Validators: []validator.Object{
+					objectvalidator.ExactlyOneOf(path.Expressions{
+						path.MatchRoot("aws"),
+						path.MatchRoot("google"),
+					}...),
+				},
+			},
+		},
+	}
+)
+
+func generateNetworkConnectionDatasourceSchema(diagn diag.Diagnostics) dataschema.Schema {
+	res := dataschema.Schema{
+		MarkdownDescription: "Network Connection datasource",
+		Attributes:          convertSchemaAttributes(networkConnectionResourceSchema.Attributes, diagn),
+	}
+
+	id := res.Attributes["id"].(*dataschema.StringAttribute)
+	id.Computed = false
+	id.Required = true
+
+	return res
+}
+
+func (m *NetworkConnectionModel) FromProtobuf(nc *network.NetworkConnection) error {
+	m.NetworkID = types.StringValue(nc.NetworkId)
+	m.ID = types.StringValue(nc.Id)
+	m.Description = types.StringValue(nc.Description)
+
+	switch {
+	case nc.GetAws() != nil:
+		if m.AWS == nil {
+			m.AWS = &awsNetworkConnectionInfo{}
+		}
+		switch {
+		case nc.GetAws().GetPeering() != nil:
+			if m.AWS.Peering == nil {
+				m.AWS.Peering = &awsNetworkConnectionPeeringInfo{}
+			}
+			params := nc.GetAws().GetPeering()
+			peering := m.AWS.Peering
+
+			peering.PeeringConnectionID = types.StringValue(params.PeeringConnectionId)
+			peering.ManagedIPv4CIDRBlock = types.StringValue(params.ManagedIpv4CidrBlock)
+			peering.ManagedIPv6CIDRBlock = types.StringValue(params.ManagedIpv6CidrBlock)
+			peering.VPCID = types.StringValue(params.VpcId)
+			peering.AccountID = types.StringValue(params.AccountId)
+			peering.RegionID = types.StringValue(params.RegionId)
+			peering.IPv4CIDRBlock = types.StringValue(params.Ipv4CidrBlock)
+			peering.IPv6CIDRBlock = types.StringValue(params.Ipv6CidrBlock)
+		default:
+			return fmt.Errorf("unsupported type of AWS connection: %v", nc.GetAws().GetType())
```

Comment:
As well, I think that error message is incorrect.
On line 251 you could have `network_connection` with `aws`-type but without `peering = nil` field.
Nothing wrong with `network_connection` type.

Correct me if I wrong.

Reply:
Right now we support the only type for AWS connection - Peering. Setting this field to nil is invalid. API could not return AWS Connection without this field. In terraform it is just a double check.

Reply:
Yeah, in this case If I were u, I would prefer to set `Required: true` to `peering` field and omit double-checks.

Reply:
This is API response validation, so we should keep it

Reply:
Decided to keep it, because API could return response which terraform doesn't support.
Valid argument, thank you for discussion